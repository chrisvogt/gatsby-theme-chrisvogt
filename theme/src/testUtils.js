import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
