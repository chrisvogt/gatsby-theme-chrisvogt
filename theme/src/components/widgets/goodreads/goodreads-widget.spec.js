import React from 'react'
import GoodreadsWidget from './goodreads-widget'
import renderer from 'react-test-renderer'
import { TestProviderWithState } from '../../../testUtils'
import useSiteMetadata from '../../../hooks/use-site-metadata'
import { Router, LocationProvider } from '@gatsbyjs/reach-router'

jest.mock('../../../hooks/use-site-metadata')

const mockSiteMetadata = {
  widgets: {
    goodreads: {
      username: 'mockusername',
      widgetDataSource: 'https://fake-api.chrisvogt.me/social/goodreads'
    }
  }
}

describe('Goodreads Widget', () => {
  beforeEach(() => {
    useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  const renderWithRouter = ui => (
    <LocationProvider
      history={{
        location: { pathname: '/' },
        listen: () => () => {},
        navigate: () => {},
        _onTransitionComplete: () => {}
      }}
    >
      <Router>{ui}</Router>
    </LocationProvider>
  )

  it('matches the loading state snapshot', () => {
    const tree = renderer
      .create(<TestProviderWithState>{renderWithRouter(<GoodreadsWidget default />)}</TestProviderWithState>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
