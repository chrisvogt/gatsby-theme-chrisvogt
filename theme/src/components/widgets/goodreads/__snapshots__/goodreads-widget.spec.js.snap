// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goodreads Widget matches the loading state snapshot 1`] = `
<section
  className="css-1pk2a5b"
  id="goodreads"
>
  <header
    className="css-gzw2dr"
  >
    <h2
      className="css-1s4ek1q"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-goodreads css-1f4q58s"
        data-icon="goodreads"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M299.9 191.2c5.1 37.3-4.7 79-35.9 100.7-22.3 15.5-52.8 14.1-70.8 5.7-37.1-17.3-49.5-58.6-46.8-97.2 4.3-60.9 40.9-87.9 75.3-87.5 46.9-.2 71.8 31.8 78.2 78.3zM448 88v336c0 30.9-25.1 56-56 56H56c-30.9 0-56-25.1-56-56V88c0-30.9 25.1-56 56-56h336c30.9 0 56 25.1 56 56zM330 313.2s-.1-34-.1-217.3h-29v40.3c-.8.3-1.2-.5-1.6-1.2-9.6-20.7-35.9-46.3-76-46-51.9.4-87.2 31.2-100.6 77.8-4.3 14.9-5.8 30.1-5.5 45.6 1.7 77.9 45.1 117.8 112.4 115.2 28.9-1.1 54.5-17 69-45.2.5-1 1.1-1.9 1.7-2.9.2.1.4.1.6.2.3 3.8.2 30.7.1 34.5-.2 14.8-2 29.5-7.2 43.5-7.8 21-22.3 34.7-44.5 39.5-17.8 3.9-35.6 3.8-53.2-1.2-21.5-6.1-36.5-19-41.1-41.8-.3-1.6-1.3-1.3-2.3-1.3h-26.8c.8 10.6 3.2 20.3 8.5 29.2 24.2 40.5 82.7 48.5 128.2 37.4 49.9-12.3 67.3-54.9 67.4-106.3z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Goodreads
    </h2>
    <div
      className="css-1rgtmy6"
    >
      <svg
        className="svg-loaders-svg css-ao632c"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-1hw7zrq"
  >
    <div
      className="css-1giak3r"
    >
       
    </div>
    <div
      className="css-zcaudb"
    >
       
    </div>
  </div>
  <div
    className="gallery"
  >
    <div
      className="css-sak8bt"
    >
      <h3
        className="css-174rnww"
      >
        Books
      </h3>
      <p
        className="css-1qwovgy"
      >
        The last 12 books I read and finished.
      </p>
      <div
        className="css-6s9lz6"
      >
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
        <div
          className="rect-shape css-1ncz0al"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="css-1yz7e9k"
  >
    <h3
      className="css-uyljfj"
    >
      Last Update
    </h3>
    <a
      className="css-1srnkuc"
    >
      <div
        className="css-dqblq9"
      >
        <div
          className="text-row show-loading-animation"
          style={
            {
              "backgroundColor": undefined,
              "height": "1em",
              "marginTop": 0,
              "maxHeight": undefined,
              "width": "100%",
            }
          }
        />
        <div
          className="css-b5lvlf"
        >
          <div
            className="text-row show-loading-animation"
            style={
              {
                "backgroundColor": "#efefef",
                "height": "15px",
                "marginTop": 0,
                "maxHeight": undefined,
                "width": "150px",
              }
            }
          />
        </div>
      </div>
    </a>
  </div>
</section>
`;
